ESB front-end modules support (#9714)
nrfconnect/sdk-nrf: doc/nrf/ug_esb.rst

The radio and timer interrupt handlers run at priority level 0 (highest level), and the ESB callback functions run at priority level 1.
Other interrupts used by the application must use priority level 2 or lower (level 2 to 7) to ensure correct operation.

> The ESB requires the :ref:`nrfxlib:mpsl_lib` for front-end modules support.

Reviewer: ```suggestion
ESB requires the :ref:`MPSL library <nrfxlib:mpsl_lib>` for front-end module support.
```

> This library is always linked into build but it is not initialized by default to give your application and the ESB protocol an access to all hardware resources.

Reviewer: ```suggestion
This library is always linked into the build, but it is not initialized by default to provide your application and the ESB protocol with access to all hardware resources.
```

> This library is always linked into the build, but it is not initialized by default to provide your application and the ESB protocol with access to all hardware resources.

Reviewer 2: ```suggestion
This library is always linked into the build, however, it must be initialized manually to provide your application and the ESB protocol with access to all hardware resources.
```

Author: I wrote it in a different way

Reviewer 2: Ok, so maybe we can split this sentence into two?

**This library is always linked to the build, yet it is not initialized by default. As a result, your application and the ESB protocol can access all hardware resources.**

> This library is always linked into the build, however, it is not initialized by default to provide your application and the ESB protocol with access to all hardware resources.

Reviewer 2: ```suggestion
This library is always linked to the build, yet it is not initialized by default. As a result, your application and the ESB protocol can access all hardware resources.
```

Reviewer: Remember, every sentence on its own line.

Reviewer 2: Corrected, thank you! :)
